os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c

```diff
@@ -339,7 +339,19 @@ static void wifi_connected_hdl( char* buf, int buf_len, int flags, void* userdat
 		(join_user_data->network_info.security_type == RTW_SECURITY_WEP_SHARED))){
 		rtw_join_status = JOIN_COMPLETE | JOIN_SECURITY_COMPLETE | JOIN_ASSOCIATED | JOIN_AUTHENTICATED | JOIN_LINK_READY | JOIN_CONNECTING;
 		rtw_up_sema(&join_user_data->join_sema);
-	}else if((join_user_data!=NULL)&&((join_user_data->network_info.security_type == RTW_SECURITY_WPA2_AES_PSK) )){
+	}else if((join_user_data!=NULL)&&((join_user_data->network_info.security_type == RTW_SECURITY_WPA2_AES_PSK) ||

```

Comment:
In this case, there are so so many pointer operations, because of `join_user_data->network_info.security_type`.
How about like below?
```
if (join_user_data != NULL) {
    rtw_security_t security_type = join_user_data->network_info.security_type;
    if (security_type == RTW_SECURITY_OPEN || security_type == AAA || ...) {
        ...
    } else if (security_type == XXX || security_type == YYY || ... ) {
```
Then we can save the time and much better to read the codes, I think.

Reply:
Dear @jeongchanKim ,

Sure I will change the code as you suggested. 
Btw, may I ask that on my pull request page, it shows "ci/circleci: checkout_code" step is failed: Error uploading workspace archive: AccessDenied: Access Denied. status code: 403.
I never encountered this issue before, may I ask if customer side add some access rights? Will this CI error affect PR merge? Do u have any idea how I can fix this?

Thank You!

Reply:
I think that is from CI problem, not from yours.
So you can ignore it :)

Reply:
ok thank you!